### ISSUE TITLE
Does not compile under gcc 8.2 after ./configure --static 

### ISSUE THREAD

**Issue by krisk0** [scylladb/seastar#512]
`./configure --static` results in `-static` flag passed to `g++` when linking:
```
g++  -O2  -I/tmp/bu/seastar/build/release/gen -I/tmp/bu/seastar/build/release/c-ares -Lbuild/release -Lbuild/release/fmt/fmt -g -Wl,--no-as-needed -static  -fvisibility=hidden  -pthread  -o build/release/apps/httpd/httpd build/release/apps/httpd/main.o build/release/http/transformers.o build/release/http/json_path.o build/release/http/file_handler.o build/release/http/common.o build/release/http/routes.o build/release/json/json_elements.o build/release/json/formatter.o build/release/http/matcher.o build/release/http/mime_types.o build/release/http/httpd.o build/release/http/reply.o build/release/http/api_docs.o build/release/net/proxy.o build/release/net/virtio.o build/release/net/dpdk.o build/release/net/ip.o build/release/net/ethernet.o build/release/net/arp.o build/release/net/native-stack.o build/release/net/ip_checksum.o build/release/net/udp.o build/release/net/tcp.o build/release/net/dhcp.o build/release/net/tls.o build/release/net/dns.o build/release/net/config.o build/release/core/reactor.o build/release/core/alien.o build/release/core/execution_stage.o build/release/core/systemwide_memory_barrier.o build/release/core/fstream.o build/release/core/posix.o build/release/core/memory.o build/release/core/resource.o build/release/core/scollectd.o build/release/core/metrics.o build/release/core/app-template.o build/release/core/thread.o build/release/core/dpdk_rte.o build/release/core/fsqual.o build/release/core/linux-aio.o build/release/util/conversions.o build/release/util/program-options.o build/release/util/log.o build/release/util/backtrace.o build/release/util/alloc_failure_injector.o build/release/net/packet.o build/release/net/posix-stack.o build/release/net/net.o build/release/net/stack.o build/release/net/inet_address.o build/release/rpc/rpc.o build/release/rpc/lz4_compressor.o build/release/core/exception_hacks.o build/release/core/future-util.o -lboost_program_options -lboost_system -lboost_filesystem -lstdc++ -lm -lstdc++fs -lboost_thread -lcryptopp -lrt -lgnutls -lgnutlsxx -llz4 -lprotobuf -ldl -lgcc_s  -lyaml-cpp -L/tmp/bu/seastar/build/dpdk/lib -Wl,--whole-archive -lrte_pmd_vmxnet3_uio -lrte_pmd_i40e -lrte_pmd_ixgbe -lrte_pmd_e1000 -lrte_pmd_ring -lrte_pmd_bnxt -lrte_pmd_cxgbe -lrte_pmd_ena -lrte_pmd_enic -lrte_pmd_fm10k -lrte_pmd_nfp -lrte_pmd_qede -lrte_pmd_sfc_efx -lrte_hash -lrte_kvargs -lrte_mbuf -lrte_ethdev -lrte_eal -lrte_mempool -lrte_mempool_ring -lrte_ring -lrte_cmdline -lrte_cfgfile -Wl,--no-whole-archive -lrt -lm -ldl -lhwloc -lnuma -lpciaccess -lxml2 -lz -lcares-seastar -lfmt  
```
Under gcc 8.2, `-static` flag does not play well with -lgcc_s:
```
/usr/lib/gcc/x86_64-pc-linux-gnu/8.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -lgcc_s
```
Don't know about other versions of gcc.

Perhaps `-lgcc_s` should be removed?

**Comment by nyh** [comment]
Yes, -lgcc_s should be dropped altogether. It is not needed - not for dynamic and definitely not for static compilation. I already noticed this in the past, and @avikivity even sent a patch for it, which I don't know why never got in: https://groups.google.com/forum/#!topic/seastar-dev/hbv3ADQUp10

The longer explanation:

We added "-lgcc_s" in commit 9ecbc9b668427cdcc9a2dd76d051b3c41c457297. As that commit explains, we needed to add it before "-lunwind" because of a bug in a specific version of libunwind. However, since commit a48fe69724492ffec64625a081f6dea54a50e470 we removed our use of libunwind (we now use backtrace() from glibc). So we don't need that "-lgcc_s" any more as well.
